packages/dds/sequence/src/test/intervalTestUtils.ts

```diff
@@ -123,16 +125,30 @@ function assertPropertiesEqual(a: SharedString, b: SharedString): void {
 		for (const key of aKeys) {
 			const aVal: unknown = aProps[key];
 			const bVal: unknown = bProps[key];
-			assert.deepEqual(
-				aVal,
-				bVal,
-				`Property sets have different values @${i} for  key ${key}: ${aVal} vs ${bVal}`,
-			);
+			if (isObject(aVal) === true) {
+				assert(isObject(bVal), `Values differ at key ${key}: ${aVal} vs ${bVal}`);
+				const aHandle = isFluidHandle(aVal) ? await aVal.get() : aVal;
+				const bHandle = isFluidHandle(bVal) ? await bVal.get() : bVal;
+				assert.equal(
+					aHandle,
+					bHandle,
+					`${a.id} and ${b.id} differ at key ${key}: ${JSON.stringify(
+						aHandle,
+					)} vs ${JSON.stringify(bHandle)}`,
+				);
+			} else {
+				assert.deepEqual(
+					aVal,
+					bVal,
+					`Property sets have different values @${i} for  key ${key}: ${aVal} vs ${bVal}`,
+				);
+			}
 		}
 		if (aKeys.length !== bKeys.length) {
```

Comment:
this also needs to handle handles. an easy option could be to delete this section and just do aKeys.concat(bKeys) in the loop above